salt/beacons/avahi_announce.py

```diff
@@ -194,12 +196,12 @@ def beacon(config):
                     grain_value = ','.join(grain_value)
             txt[item] = _enforce_txt_record_maxlen(item, grain_value)
             if LAST_GRAINS and (LAST_GRAINS.get(grain, '') != __grains__.get(grain, '')):
-                changes[str('txt.' + item)] = txt[item]
+                changes[six.text_type('txt.' + item)] = txt[item]
```

Comment:
It looks like the `str()` was here unnecessarily in the first place. I'm assuming that `item` is always a string here, otherwise the concatenation of `'txt.'` and `item` would fail:

```
>>> x = 9
>>> 'txt.' + x
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: cannot concatenate 'str' and 'int' objects
>>>
```

The one thing that could potentially be a problem here is if `item` contains unicode characters, but encoded into a `str` type.

What we should do here is add the following line right below the `for` statement on line 182:

```python
changes_key = 'txt.' + salt.utils.stringutils.to_unicode(item)
```

Then we can change lines 199 and [204](https://github.com/saltstack/salt/pull/45238/files#diff-024253fb5f90e110e541d733210f45c1R204) to the following:

```python
changes[changes_key] = txt[item]
```